frontend/src/component/admin/network/NetworkTrafficUsage/NetworkTrafficUsage.tsx

```diff
@@ -225,65 +231,42 @@ const NewNetworkTrafficUsage: FC = () => {
             },
             includedTraffic,
         );
-    }, [theme, newPeriod]);
+    }, [theme, chartDataSelection]);
 
-    const traffic = useInstanceTrafficMetrics2(newPeriod);
+    const traffic = useInstanceTrafficMetrics2(
+        chartDataSelection.grouping,
+        toDateRange(chartDataSelection),
+    );
 
     const data = newToChartData(traffic.usage);
+    const usageTotal = calculateTotalUsage(traffic.usage);
+    const overageCost = calculateOverageCost(
+        usageTotal,
+        includedTraffic,
+        BILLING_TRAFFIC_BUNDLE_PRICE,
+    );
```

Comment:
We don't need all the `useState` stuff to calculate this, so I've changed it to just regular functions.

Reply:
The reason for useStates littered across here is some uncertainty in whether or not a variable would be updated when dependencies updated. This might not be a problem, but potentially something to be aware of

Reply:
yeah. I've not seen any issues with that. I'll double check and then again when this goes out to sandbox, but I think it's fine.